server/interceptor/auth.go

```diff
@@ -0,0 +1,41 @@
+package interceptor
+
+import (
+	"context"
+	"githubsearchservice/pkg"
+	"log"
+	"strings"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+)
+
+func Auth() grpc.UnaryServerInterceptor {
+	return func(
+		ctx context.Context,
+		req interface{},
+		info *grpc.UnaryServerInfo,
+		handler grpc.UnaryHandler,
+	) (interface{}, error) {
+		log.Println("incoming request: ", info.FullMethod, req)
+
+		md, ok := metadata.FromIncomingContext(ctx)
+		if !ok {
+			return nil, status.Errorf(codes.Unauthenticated, "metadata is not provided")
+		}
+
+		values := md["github_api_token"]
```

Comment:
The metadata key should probably be a constant since it shared in many files.
```suggestion
		values := md["github_api_token"]
```